qiskit/qasm3/exporter.py

```diff
@@ -448,15 +449,33 @@ def build_quantumdeclarations(self):
         )
         # aliases
```

Comment:
Not a request for changes, but just wondering if I'm understanding this correctly. Does this (very rough) pseudocode capture the idea?

```py
while qubit:
    range_bits = qubit.take(1).takeWhile(lambda bit: bit_location[bit] == bit_location[magic_prev_bit] + 1)
    qubit = qubit.skip(len(range_bits))
    elements.append(SI(range_bits))
```

Reply:
Yeah, looks like it.  Most of the bookkeeping is just around setting up the first indices and ensuring that the final range is emitted too.